### PR TITLE
Made some changes and broke something

### PR DESCRIPTION
**Pull request by Zrp200**
- states are now represented as symbols
- Please fix `Game#play` before merging
  - I can't get it to exit loop
- Fixed spelling mistakes
- changed some `while`s to `until`s
- combined `Grid#new` and `Grid#build`
- removed unneeded `()`s

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 # Game class. Main program class.
 class Game
   attr_reader :state
 
-  STATES = %w(initialized ready error terminated gameover).freeze
+  STATES = %i(initialized ready error terminated game_over)
   GRID_SIZE = 10
-  HIT_CHAR = 'X'.freeze
-  MISS_CHAR = '-'.freeze
-  NO_SHOT_CHAR = '·'.freeze
```
Comment (szymon33) [inline lib/game.rb:12]: Why did you remove `freeze` ?

File: lib/grid.rb [hunk 2]
```diff
@@ -2,28 +2,20 @@
 class Grid
   attr_accessor :status_line
 
-  AXE_LETTERS = %w( A B C D E F G H I J ).freeze
-  AXE_DIGGITS = %w( 1 2 3 4 5 6 7 8 9 10 ).freeze
-  HIT_CHAR = 'X'.freeze
-  MISS_CHAR = '-'.freeze
```
Comment (szymon33) [inline lib/grid.rb]: Same question as above. Why did you remove freeze?

File: lib/grid.rb [hunk 3]
```diff
@@ -2,28 +2,20 @@
...
-
-  def initialize
-    @matrix = []
-    @fleet = []
-  end
+  AXE_LETTERS = %w( A B C D E F G H I J )
+  AXE_DIGITS = %w( 1 2 3 4 5 6 7 8 9 10 )
+  HIT_CHAR = 'X'
+  MISS_CHAR = '-'
```
Comment (szymon33) [inline lib/grid.rb:8]: Same question as above. Why did you remove freeze?

File: lib/grid.rb [hunk 4]
```diff
@@ -2,28 +2,20 @@
...
-    self
   end
 
   def show
     print_header
     setup_with_fleet if @fleet
-    @matrix.each_with_index do |grow, index|
-      Grid.row("#{AXE_LETTERS[index]} #{grow.join(' ')}")
-    end
+    @matrix.each_with_index {|grow, index| Grid.row("#{AXE_LETTERS[index]} #{grow.join(' ')}")}
```
Comment (szymon33) [inline lib/grid.rb:18]: Yes, it could be in one line but for me it is harder to read now. I think previous version was more readable.

File: lib/grid.rb [hunk 5]
```diff
@@ -37,17 +29,15 @@ def self.row(txt)
   def setup_with_fleet
     if @fleet
       @fleet.each do |ship|
-        ship.location.each do |coordinates|
-          @matrix[coordinates[0]][coordinates[1]] = HIT_CHAR
-        end
+        ship.location.each {|coordinates| @matrix[coordinates.first][coordinates[1]] = HIT_CHAR}
```
Comment (szymon33) [inline lib/grid.rb:32]: I care about Ruby style. Please use space before and after `{` Take a look at: https://github.com/bbatsov/ruby-style-guide#spaces-operators . But I would keep a previous version as more readable (shorter line, less in-chain).

File: lib/grid.rb [hunk 6]
```diff
@@ -37,17 +29,15 @@ def self.row(txt)
...
       end
     end
     @matrix
   end
 
   def print_header
-    Grid.row('=' * AXE_DIGGITS.size * 3)
-    Grid.row(status_line)
-    Grid.row("  #{AXE_DIGGITS.join(' ')}")
+    Grid.row(?= * AXE_DIGITS.size * 3)
```
Comment (szymon33) [inline lib/grid.rb:39]: This is interesting :smile:

File: lib/ship.rb [hunk 7]
```diff
@@ -38,36 +38,34 @@ def build
...
+    @location.push xy
     @matrix[xy[0]][xy[1]] = true
   end
 
   # returns valid surrounding mask
   def take_mask(xy, exception = nil)
-    return [] unless xy
-    x = xy[0]
+    return Array.new unless xy
+    x = xy.first
```
Comment (szymon33) [inline lib/ship.rb:52]: I would keep previous version because the context is more readable like: `xy[0], xy[1], ...`

File: lib/ship.rb [hunk 8]
```diff
@@ -38,36 +38,34 @@ def build
...
     mask[1] = [x    , y - 1] if (y - 1) >= 0
     mask[2] = [x    , y + 1] if (y + 1) < @matrix.size
     mask[3] = [x + 1, y    ] if (x + 1) < @matrix.size
-    clean(mask, exception)
+    clean mask, exception
   end
 
   def clean(mask, exception)
-    mask = mask.select { |item| item && item != exception }
+    mask.reject! { |item| item || item == exception }
```
Comment (szymon33) [inline lib/ship.rb:66]: Good catch.

File: lib/ship.rb [hunk 9]
```diff
@@ -38,36 +38,34 @@ def build
...
 
-    mask.each do |item|
-      return [] if @matrix[item[0]][item[1]] == true
-    end
+    mask.each {|item| return Array.new if @matrix[item[0]][item[1]]}
```
Comment (szymon33) [inline lib/ship.rb:68]: Ruby style. Should be space before and after `{`  `}`

File: spec/lib/ship_spec.rb [hunk 10]
```diff
@@ -31,7 +31,7 @@
       expect(ship).to respond_to(:build)
     end
 
-    it 'makes ship array' do
+    it 'makes ship an Array' do
```
Comment (szymon33) [inline spec/lib/ship_spec.rb:34]: Thank you for your collaboration Zachary and nice to meet you!

**Comment by Zrp200** [comment]
I removed freeze since you can't change constants

File: lib/ship.rb (hunk 7 above)
Comment (Zrp200) [inline lib/ship.rb:52]: I disagree; words are easier to understand than symbols

**Comment by szymon33** [comment]
Are you sure about it?

Please consider the following code:

``` ruby
class HowConstantWork
  AB = 'foo'
end

puts HowConstantWork::AB
# foo
mutant = HowConstantWork::AB
mutant << 'bar'
puts HowConstantWork::AB
# foobar
```

I appreciate your collaboration, but could we agree to keep `freeze` here?

**Comment by Zrp200** [comment]
Ruby complains when you change it

**Comment by Zrp200** [comment]
What did I do to break `Game#play`??? :disappointed:

File: lib/game.rb [hunk 11]
```diff
@@ -100,7 +97,7 @@ def console
   end
 
   def shoot
-    if xy = convert
+    if xy == convert
```
Comment (szymon33) [inline lib/game.rb:100]: No man. This was intended to be assignment and comparison in one step. Did you actually play in this game after you made your changes?

**Comment by szymon33** [comment]
> What did I do to break Game#play??? 

I fixed everything in the following commit : https://github.com/szymon33/battleship/commit/77c59f5f1db36d406210bb9d73b6dde65fcbde11

**Comment by Zrp200** [comment]
thanks
